modules/stochastic_tools/src/state_sim/model/events/StateSimEvCodeEval.C

```diff
@@ -0,0 +1,39 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "StateSimEvCodeEval.h"
+#include "StateSimModel.h"
+#include <string>
+
+StateSimEvCodeEval::StateSimEvCodeEval(std::string name)
+  : StateSimEvConditionBased(name, ET_CODE_EVAL),
+    _code_eval(NULL)
+{
+}
+
+StateSimEvCodeEval::StateSimEvCodeEval(std::string name, CodeEvalFunc code_eval, std::vector<StateSimVariable> * watch_list)
+  : StateSimEvConditionBased(name, ET_CODE_EVAL),
+    _code_eval(code_eval)
+{
+  if (watch_list != NULL)
+  {
+    for (auto var : *watch_list)
+    {
+      this->addRelatedItem(var);
+    }
+  }
+}
+
+bool
+StateSimEvCodeEval::isTriggered(const TimespanH & cur_ev_time, const StateSimBitset & cur_states, const StateSimModel & main_model)
+{
+  if (_code_eval == NULL)
+    mooseAssert(false, "StateSimEvCodeEval isTriggered() - code_eval not assigned"); //done in if for debuging
```

Comment:
mooseAssert shouldn't be the only statement in the `if` branch. Move the test to the assert